src/regtech_data_validator/check_functions.py

```diff
@@ -39,6 +39,21 @@ def _check_blank_(value: str, check_result: bool, accept_blank: bool = False) ->
         return check_result
 
 
+def comparison_helper(value: str, sign: str, limit: str, accept_blank: bool = False) -> bool:
+    """
+    Used to first check if blank values are acceptable and returns corresponding boolean result.
+    Then, if actual value exists, uses python eval() to convert string to numerical values
+    and then evals again to use the passed in sign to compare the value to the limit.  This
+    ensures accurate comparison checks.
+    """
+    if accept_blank and not value.strip():
+        return True
+    elif not accept_blank and not value.strip():
+        return False
+    else:
+        return (bool)(eval(f"eval(value) {sign} eval(limit)"))
```

Comment:
not a fan of using evals in general, why can't we just cast them to int or float as I believe eval would run into the same issues with non-numerical strings?
also, I think having this just be the blank check would be enough, and have the actual comparison logic in the original functions that are named as the logic.
if we want to shortcut the comparison, then we can just have this blank_checker return True, False, or None, and if it's None, then do the logical comparison.

Reply:
The original intent for the phase 2 validations was to convert all the data to be typed appropriately, and then have all the phase 2 check functions be typed as well. Then you wouldn't have to redo all the string parsing logic/handling over again that the phase 1 validations already did.

That work obviously never happed, but I still like the idea. I know it doesn't help here, but  something we should evaluate post-MVP.

Reply:
> not a fan of using evals in general, why can't we just cast them to int or float as I believe eval would run into the same issues with non-numerical strings? also, I think having this just be the blank check would be enough, and have the actual comparison logic in the original functions that are named as the logic. if we want to shortcut the comparison, then we can just have this blank_checker return True, False, or None, and if it's None, then do the logical comparison.

Was using eval for future cases where we might have ints or floats, and if we do a conversion to either, all three functions would need to implement the same code so I put it in one place.   Each would need to check for existence, then the specific type, then convert and then do the comparison.  The existing blank checker checked for existence afterwards and with casting, we have to check all that before.